scripts/validate_docstrings.py

```diff
@@ -497,6 +498,20 @@ def parameter_desc(self, param):
                 desc = desc[: desc.index(full_directive)]
         return desc
 
+    def has_proper_punctuation(self, param):
+        """Return True if a parameter description is terminated correctly.
+
+        A parameter description should always be terminated with a period, a
+        part from when it ends with a bullet point.
+        """
+        if self.parameter_desc(param)[-1] == ".":
+            return True
+        else:
+            # Check if the parameter description ends with a bullet list
+            if re.search(BULLET_LIST_PATTERN, self.parameter_desc(param)):
```

Comment:
Are directives included in `self.parameter_desc`? The way this is written it really just checks if there is a bullet point somewhere in the description, but without regarding the position. I thought something with `re.findall` would be more appropriate just not sure how complicated that gets with directives

Reply:
Hi @WillAyd , thank you for this and apologies again for the late reply.
`self.parameter_desc` returns the parameter docstring as 1 string removing everything after a directive.

For example, this method:
```
method : {'min', 'max'}, default 'min'
   Parameter introduction:

   * 'min': the description for min
   * 'max': the description for max

   .. versionadded:: 0.1.2
   .. deprecated:: 0.00.0
        A description
```
will be simply returned as 
`"Parameter introduction:* 'min': the description for min* 'max': the description for max"`

Since the newline characters (and blank lines) are also lost using `self.parameter_desc`, I am finding tricky to come up with a regex that perfectly matches a bullet list. Do you have anything in mind?